homeassistant/components/zwave_me/fan.py

```diff
@@ -0,0 +1,70 @@
+"""Representation of a fan."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ZWaveMeEntity
+from .const import DOMAIN, ZWaveMePlatform
+
+DEVICE_NAME = ZWaveMePlatform.FAN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the fan platform."""
+
+    @callback
+    def add_new_device(new_device):
+        controller = hass.data[DOMAIN][config_entry.entry_id]
+        switch = ZWaveMeFan(controller, new_device)
+
+        async_add_entities(
+            [
+                switch,
+            ]
+        )
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, f"ZWAVE_ME_NEW_{DEVICE_NAME.upper()}", add_new_device
+        )
+    )
+
+
+class ZWaveMeFan(ZWaveMeEntity, FanEntity):
+    """Representation of a ZWaveMe Fan."""
+
+    _attr_supported_features = FanEntityFeature.SET_SPEED
+
+    @property
+    def percentage(self) -> int:
+        """Return the current speed as a percentage."""
+        return self.device.level
```

Comment:
Is the range 0-99 or 0-100?

Reply:
Do you mean we should rescale out 0..99 to 0..100?

Reply:
If the level range is 0..99, maybe something like this:

```py
from homeassistant.util.percentage import ranged_value_to_percentage

...

    @property
    def percentage(self) -> int:
        """Return the current speed as a percentage."""
        if (level := self.device.level) == 0:
            return 0

        return ranged_value_to_percentage((0, 99), level)
```